crates/uv-resolver/src/preferences.rs

```diff
@@ -115,6 +115,21 @@ impl Preference {
         }))
     }
 
+    /// Create a [`Preference`] from an installed distribution.
+    pub fn from_installed(dist: &InstalledDist) -> Option<Self> {
```

Comment:
Note from reviewing: We need this separately from the `InstalledPackagesProvider`  in `CandidateSelector` to have a shared preferences type between preferences from lock and preferences from installed.